File: python/raindrops/raindrops.py

DROP_TYPES = [
    (3, 'Pling'),
    (5, 'Plang'),
    (7, 'Plong'),
]


def convert(n):
    result = [sound for factor, sound in DROP_TYPES if n % factor == 0]
    return ''.join(result) or str(n)

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert takes a raindrop number returns a string of the sounds.
func Convert(n int) string {

	var sounds string

	if n%3 == 0 {
		sounds += "Pling"
	}

	if n%5 == 0 {
		sounds += "Plang"
	}

	if n%7 == 0 {
		sounds += "Plong"
	}

	if sounds == "" {
		sounds = strconv.Itoa(n)
	}

	return sounds

}
